pkg/api/v1/defaults.go

```diff
@@ -389,3 +389,32 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+func SetDefaults_SecurityContext(sc *v1.SecurityContext) {
+	pt := true
+	if sc == nil {
+		sc = &v1.SecurityContext{
+			AllowPrivilegeEscalation: &pt,
+		}
+		return
+	}
+
+	// handle the case for AllowPrivilegeEscalation when the container is privileged
+	if sc.Privileged != nil && *sc.Privileged {
+		sc.AllowPrivilegeEscalation = &pt
+	}
+
+	// handle the case for AllowPrivilegeEscalation when we are adding CAP_SYS_ADMIN
+	if sc.Capabilities != nil {
+		for _, cap := range sc.Capabilities.Add {
+			if string(cap) == "CAP_SYS_ADMIN" {
+				sc.AllowPrivilegeEscalation = &pt
+			}
+		}
+	}
+
+	// handle the case where the user did not set the defaultAllowPrivilegeEscalation and did not explicitly set allowPrivilegeEscalation
+	if sc.AllowPrivilegeEscalation == nil {
```

Comment:
Comment assumes that `defaultAllowPrivilegeEscalation == nil` but code doesn't check for that. Is it intentionally?

Reply:
yeah I was just trying to relay that the securityContext would be nil if they didn't set a default.